addons/mail/static/src/discuss/message_pin/common/pinned_messages_panel.js

```diff
@@ -21,28 +19,16 @@ export class PinnedMessagesPanel extends Component {
     static template = "discuss.PinnedMessagesPanel";
 
     setup() {
-        this.store = useService("mail.store");
-        this.messagePinService = useState(useService("discuss.message.pin"));
         onWillStart(() => {
-            this.messagePinService.fetchPinnedMessages(this.props.thread);
+            this.props.thread.fetchPinnedMessages();
         });
         onWillUpdateProps(async (nextProps) => {
```

Comment:
Was already in code, but this `async` could be removed as it looks useless.
`async` can even introduce bugs, because an `async` function adds a microtask that could lead to state inconsistency if we are not cautious enough. Considering that OWL still has some problems in that domain, better avoid adding `async` that could break code easily